Tasks/Common/artifacts-common/telemetry.ts

```diff
@@ -0,0 +1,50 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as semver from 'semver';
+
+/**
+ * Utility function to log telemetry.
+ * @param feature The task/feature name for this telemetry
+ * @param telem A JSON object containing a dictionary of variables that will be appended to
+ * common system vars and loggged.
+ */
+export function emitTelemetry(area: string, feature: string, taskSpecificTelemetry: any) {
+    try {
+        let agentVersion = tl.getVariable('Agent.Version');
+        if (semver.gte(agentVersion, '2.120.0')) {
+            // Common Telemetry VARs that will be concatenated with the supplied telem object.
+            let commonTelem = {
+                'SYSTEM_TASKINSTANCEID': tl.getVariable('SYSTEM_TASKINSTANCEID'),
+                'SYSTEM_JOBID': tl.getVariable('SYSTEM_JOBID'),
+                'SYSTEM_PLANID': tl.getVariable('SYSTEM_PLANID'),
+                'SYSTEM_COLLECTIONID': tl.getVariable('SYSTEM_COLLECTIONID'),
+                'AGENT_ID': tl.getVariable('AGENT_ID'),
+                'AGENT_MACHINENAME': tl.getVariable('AGENT_MACHINENAME'),
+                'AGENT_NAME': tl.getVariable('AGENT_NAME'),
+                'AGENT_JOBSTATUS': tl.getVariable('AGENT_JOBSTATUS'),
+                'AGENT_OS': tl.getVariable('AGENT_OS'),
+                'AGENT_VERSION': tl.getVariable('AGENT_VERSION'),
+                'BUILD_BUILDID': tl.getVariable('BUILD_BUILDID'),
+                'BUILD_BUILDNUMBER': tl.getVariable('BUILD_BUILDNUMBER'),
+                'BUILD_BUILDURI': tl.getVariable('BUILD_BUILDURI'),
+                'BUILD_CONTAINERID': tl.getVariable('BUILD_CONTAINERID'),
+                'BUILD_DEFINITIONNAME': tl.getVariable('BUILD_DEFINITIONNAME'),
+                'BUILD_DEFINITIONVERSION': tl.getVariable('BUILD_DEFINITIONVERSION'),
+                'BUILD_REASON': tl.getVariable('BUILD_REASON'),
+                'BUILD_REPOSITORY_CLEAN': tl.getVariable('BUILD_REPOSITORY_CLEAN'),
+                'BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT': tl.getVariable('BUILD_REPOSITORY_GIT_SUBMODULECHECKOUT'),
+                'BUILD_REPOSITORY_NAME': tl.getVariable('BUILD_REPOSITORY_NAME'),
+                'BUILD_REPOSITORY_PROVIDER': tl.getVariable('BUILD_REPOSITORY_PROVIDER'),
+                'BUILD_SOURCEVERSION': tl.getVariable('BUILD_SOURCEVERSION')
```

Comment:
We can probably trim down this list a little - e.g. I don't think we get any value from all the BUILD_REPOSITORY values

Reply:
I thought about pruning it but seemed potentially contentious for some of them so I didn't. On second thought I guess it's really just those and BUILD_SOURCEVERSION that we don't care about, though?

Reply:
Maybe BUILD_REASON too.
We could add in Agent_OSArchitecture

Reply:
I think the set of variables (and other environment info) that is versus isn't useful is a very worthwhile discussion, but should probably be separate from this PR. I'm happy to make the above changes if you'd like, but would like to hold off on any other such changes in this PR.

Reply:
fwiw I think BUILD_REASON is still potentially interesting to know which scenarios (PR, CI) package management is being used in, but definitely not critical.